src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -54,10 +55,13 @@ class Dockerfile extends DefaultTask {
     @OutputFile
     final RegularFileProperty destFile
 
+    private final ObjectFactory objects
+
     Dockerfile() {
         instructions = project.objects.listProperty(Instruction).empty()
         destFile = project.objects.fileProperty()
         destFile.set(project.layout.buildDirectory.file('docker/Dockerfile'))
+        objects = project.objects
```

Comment:
Can you explain why this is a problem with the configuration cache exactly?

Reply:
So in the [configuration cache guide](https://docs.gradle.org/7.4.2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution) one of the requirements is that `Task.getProject()` is disallowed while the task is running. In this particular case, we fall under the [injected services](https://docs.gradle.org/7.4.2/userguide/custom_gradle_types.html#service_injection) category.

Reply:
The test while running without the fix fails with
```
* What went wrong:
Configuration cache problems found in this build.

4 problems were found storing the configuration cache, 1 of which seems unique.
- Task `:dockerfile` of type `com.bmuschko.gradle.docker.tasks.image.Dockerfile`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/7.4.2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```